cips/cip-blob-verified-signer.md

```diff
@@ -0,0 +1,92 @@
+- For each blob, reconstruct the blob commitment.
+- Fetch the PFB namespace
+- Parse the PFB namespace and create a mapping from blob commitment -> PFB
+- (In zero-knowledge) Accept the list of all rollup blobs and the list of relevant PFBs as inputs
+- (In zero-knowledge) Verify that the claimed list of blobs matches the block header using the namespaced merkle tree
+- (In zero-knowledge) For each blob, find the PFB with the matching commitment and check that the sender is correct.
+- (In zero-knowledge) For each relevant PFB, check that the bytes provided match the namespaced merkle tree
+
+This is currently a needlessly complicated flow and more computationally heavy at constructing proofs. This CIP proposes an easier path for rollups that opt for this fork-choice rule
+
+## Specification
+
+This CIP introduces a new blob type (known henceforth as a v2 blob given the share format version change):
+
+```proto
+message Blob {
+  bytes namespace_id = 1;
+  bytes data = 2;
+  uint32 share_version = 3;
+  uint32 namespace_version = 4;
+  // new field
+  string signer = 5;
+}
+```
+
+Given proto's backwards compatibility, users could still submit the old blob type (in the `BlobTx` format) and signer would be processed as an empty string.
+
+The new block validity rule (In `PrepareProposal` and `ProcessProposal`) would thus be that if the signer was not empty, then it must match that of the PFB that paid for it. When validating the `BlobTx`, validators would check the equivalency of the PFB's `signer` to the Blob's `signer` (as well as verification of the signature itself).
+
+Although no version changes are required for protobuf encoded blobs, share encoding would change. Blobs containing a non empty signer string would be encoded using the new v2 format:
+
+![Diagram of V2 Share Format](../assets/cip-blob-verified-signer/blob-v2-share-format.svg)
+
+Blobs with an empty `signer` string would remain encoded using the v1 format. Note that in this diagram it is the `Info Byte` that contains the share version. Not to be confused with the namespace version.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+## Rationale
+
+Given the current specification change, the new loop is simplified:
+
+- Retrieve all blobs in the subscribed namespaces
+- Verify the namespaces inclusion
+- Verify that the `signer` in each blob matches that of the sequencer
+
+As a small digression, it may be feasible to additionally introduce a new namespace version with the enforcement that all blobs in that namespace use the v2 format i.e. have a signer. However, this does not mean that the signer matches that of the sequencer (which Celestia validators would not be aware of). This would mean that full nodes would need to get and verify all blobs in the namespace anyway.
```

Comment:
This is a good call out. If Celestia also added an RPC endpoint for fetching blob metadata, this could be made quite efficient. Rollups would just have to...
1. Fetch the metadata via RPC
2. For each blob from an allowed sender, fetch the full blob and its proof via RPC
3. (zk-rollups only) For each other blob, fetch *only the first share* of the blob and its proof via RPC to check that the metadata is correct

Reply:
Iiuc, you mean that there would be a RPC query to fetch blobs by namespace and signer?

Reply:
So another way of framing this would be the minimal set of data for the fork-choice rule to filter out what blobs are worth processing or not. Wouldn't this data need to be provable? Wouldn't this make it look similar to the `PFBNamespace` although a little more compact

Reply:
This data is already provable, since it's encoded in the start share of each blob anyway. But TBC, we at Soveregin don't actually need/want this - I was just pointing it out in case it's useful for optimistic rollups

Reply:
> `The location of the blob. Alternatively, we could put the blob commitment`
I think that this is not going to be the most atomic way to point to a blob. If I submit 2 blobs that are identical, their commitments will be the same, and their index will also be different, although they have the same commitment.